### ISSUE TITLE
error: conflicting types for 'SDL_main'

### ISSUE THREAD

**Issue by dedosmedia** [Genymobile/scrcpy#1209]
I have been trying to compile scrcpy from the sources and building it as debug. (From windows MSYS2)

```
meson debug --buildtype debug --strip -Db_lto=true -Dportable=true -Dcompile_app=true -Dcompile_server=true -Dserver_debugger=true

ninja -C debug
```
But this is what I get:

```
...
BUILD SUCCESSFUL in 2s
28 actionable tasks: 28 up-to-date
[1/3] Compiling C object app/a172ced@@test_cli@exe/tests_test_cli.c.obj.
FAILED: app/a172ced@@test_cli@exe/tests_test_cli.c.obj
cc @app/a172ced@@test_cli@exe/tests_test_cli.c.obj.rsp
<command-line>: error: conflicting types for 'SDL_main'
../app/tests/test_cli.c:122:5: note: in expansion of macro 'main'
  122 | int main(void){
      |     ^~~~
In file included from C:/msys64/mingw64/include/SDL2/SDL.h:32,
                 from ../app/src/screen.h:5,
                 from ../app/src/input_manager.h:11,
                 from ../app/src/scrcpy.h:8,
                 from ../app/src/cli.h:7,
                 from ../app/tests/test_cli.c:3:
C:/msys64/mingw64/include/SDL2/SDL_main.h:121:29: note: previous declaration of 'SDL_main' was here
  121 | extern SDLMAIN_DECLSPEC int SDL_main(int argc, char *argv[]);
      |                             ^~~~~~~~
ninja: build stopped: subcommand failed.

```

I am able to compile it changing all test_*.c files

From this:
```
int main(void){ 
```
To this:
```
int main(int argc, char *argv[]){
```
Why is this issue happening to me?
Is there a way to avoid making this change to the tests?

**Comment by rom1v** [comment]
>     <command-line>: error: conflicting types for 'SDL_main'
>     ../app/tests/test_cli.c:122:5: note: in expansion of macro 'main'
>      122 | int main(void){

That's unexpected. If `SDL_MAIN_HANDLED` is defined, then SDL does not redefine `main` to `SDL_main`:
https://github.com/spurious/SDL-mirror/blob/180e65dc2a51146de7604aafb53fe4e037dea923/include/SDL_main.h#L33-L110

(could you check your `include/SDL_main.h`?)

And for tests, we define `SDL_MAIN_HANDLED`:
https://github.com/Genymobile/scrcpy/blob/c396758b4ed29cb4d4ad51e8bcdd1deecf5761d2/app/meson.build#L180

**Comment by rameshbo** [comment]
@dedosmedia 

I came across the exact issue. What did you do to fix it?

**Comment by rom1v** [comment]
@rameshbo On which version/commit?

**Comment by rameshbo** [comment]
Thanks @rom1v for the response. It is v1.14.

BTW. Wonderful piece of code. Amazing work.

**Comment by rom1v** [comment]
> It is v1.14.

Arf, so there is still a problem with this stuff.

 - ffa8c66979d4262a95cad584894b9a5fbeea0e84
 - 3259c60b22598379e86fb4e20352f95f3282d0af
 - ba1b36758e8a6cbebe8215d9bb0936d4598d6045

If you find a way to fix it (so that it works on _all_ platforms (including mingw for Linux, mingw from Windows, natively from Windows), please propose a patch :wink:

**Comment by rameshbo** [comment]
It was this one: https://github.com/Genymobile/scrcpy/commit/3259c60b22598379e86fb4e20352f95f3282d0af

I would love to, though I am not yet in a stage to propose a patch that would work in all platforms.

Currently I am trying out with mingw from Windows.

**Comment by rameshbo** [comment]
@rom1v 

I was able to fix for Window platform by using "int main(int argc, char *argv[]){" in tests instead of "int main(void) {"

can we not use int main(int argc, char *argv[]){ for tests as well?

It makes sense as I was reading this at some site: "It's important that we have the arguments of the function be an integer followed by a char* array and the return type be an integer. Any other type of main function will cause an undefined reference to main. SDL requires this type of main so it is compatible with multiple platforms."

**Comment by rom1v** [comment]
> can we not use int main(int argc, char *argv[]){ for tests as well?

Yes, we can :+1:

Could you open a PR with that change, please?

**Comment by rom1v** [comment]
Fixed by 199c74f62f5f78748769e6489122242a3217dc54.
